azerothcore/azerothcore-wotlk
# fix(Core/Globals): Check if creatures got any valid `InhabitType` dur…

…ing load to prevent broken creatures



## Changes Proposed:
-  Check if creatures got any valid `InhabitType` during load to prevent broken creatures from going unnoticed.
-  We are currently only checking if InhabyType = 0 or InhabitType = absurd, but we gotta also check if the creature actually has any valid inhabittype
- This is an easy mistake to make but an uncommon one, and as an uncommon one when it happens it's usually not spotted right away (e.g https://github.com/azerothcore/azerothcore-wotlk/pull/7426#issuecomment-941741277)
- The goal here is not that the creature works in-game if a broken template is loaded, but that the user is alerted that wrong data was used and he should see it fixed

## Issues Addressed:

- Closes 

## SOURCE:


## Tests Performed:

- Tested starting up with intentionally messed up templates.
- Tested in-game.

Creatures that only got InhabitType 8 currently will reset once pulled, as they never find a valid path to their target.

I applied the default correction here, since I saw that was used in the other cases of wrong inhabittype, but the result is still a creature behaving very badly.


## How to Test the Changes:


```sql
UPDATE `creature_template` SET `InhabitType` = 8 WHERE (`entry` = 30455); -- Previously 1
```

1. run the query above
2. start the server
3. check db logs
4. revert the broken query

```sql
UPDATE `creature_template` SET `InhabitType` = 1 WHERE (`entry` = 30455); -- Previously 1
```

## Known Issues and TODO List:




## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
